### ISSUE TITLE
[BUG] Stack Overflow error on attempt to do lookup entity (stream)

### ISSUE THREAD

**Issue by xwrs** [Permify/permify#681]
Service crashes with panic of stack overflow. Check permission requests go through. Few requests go through and fail with EOF error (probably expected).  Then subsequent requests fail because service is down. errors:

```
Fine-grained Authorization System v0.5.2
{"level":"info","time":"2023-09-18T15:49:25Z","caller":"/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940","message":"🚀 starting permify service..."}
2023/09/18 15:49:25 goose: no migrations to run. current version: 20230729145034
{"level":"info","time":"2023-09-18T15:49:25Z","caller":"/home/runner/work/permify/permify/pkg/cmd/serve.go:305","message":"🚀 grpc server successfully started: 3478"}
{"level":"info","time":"2023-09-18T15:49:25Z","caller":"/home/runner/work/permify/permify/pkg/cmd/serve.go:305","message":"🚀 http server successfully started: 3476"}
runtime: goroutine stack exceeds 1000000000-byte limit
runtime: sp=0xc0209f8370 stack=[0xc0209f8000, 0xc0409f8000]
fatal error: stack overflow

runtime stack:
runtime.throw({0x152a92a?, 0x20aaac0?})
	/opt/hostedtoolcache/go/1.20.8/x64/src/runtime/panic.go:1047 +0x5d fp=0xc000071e18 sp=0xc000071de8 pc=0x436fbd
runtime.newstack()
	/opt/hostedtoolcache/go/1.20.8/x64/src/runtime/stack.go:1105 +0x5bd fp=0xc000071fc8 sp=0xc000071e18 pc=0x451a5d
runtime.morestack()
	/opt/hostedtoolcache/go/1.20.8/x64/src/runtime/asm_amd64.s:574 +0x8b fp=0xc000071fd0 sp=0xc000071fc8 pc=0x46b1eb

```
**To Reproduce**
Create authorization model leveraging hierarchical relations. Create tuples. Query lookup of entities.
Depth parameter of request metadata is set to 3

Here is a sample of my code and/or tests that demonstrate the issue in my app:

```
listObjectsStream, err := s.permifyClient.Permission.LookupEntityStream(server.Context(), &v1.PermissionLookupEntityRequest{
		TenantId: request.TenantId,
		Metadata: &v1.PermissionLookupEntityRequestMetadata{
			SchemaVersion: request.SchemaVersion,
			Depth:         3,
		},
		EntityType: request.ResourceType,
		Permission: request.Relation,
		Subject: &v1.Subject{
			Type: request.SubjectType,
			Id:   request.SubjectId,
			//Relation: "",
		},
		//Context:    nil, //TODO: ???!!!
	})
```

**Expected behavior**
Lookup query returns data

**Additional context**
Model

```
entity user {}
entity tag {
    relation assignee @department
    permission view_document = assignee.view_document
}
entity document {
    relation owner @department

    permission edit = owner.edit_document
    permission view = owner.view_document or owner.peek_document
}

entity department {
    relation parent @department
    relation admin @user
    relation viewer @user
    relation assigned_tag @tag
    
    permission peek_document = assigned_tag.view_document or parent.peek_document
    permission edit_document = admin or parent.edit_document
    permission view_document = viewer or admin or parent.view_document
}


```

**Environment (please complete the following information, because it helps us investigate better):**
 - k8s
 - postgres 15

**Comment by tolgaOzen** [comment]
Hi @xwrs , we've resolved that issue in version v0.5.3. Please check if it's working correctly for you now. If so, could you please close the issue?

**Comment by xwrs** [comment]
@tolgaOzen thank you for a prompt fix. issue is no longer there. cheers
